server/src/main/java/org/opensearch/index/snapshots/blobstore/BlobStoreRemStoreBasedIndexShardSnapshot.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.snapshots.blobstore;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentParserUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Remote Store based Shard snapshot metadata
+ *
+ * @opensearch.internal
+ */
+public class BlobStoreRemStoreBasedIndexShardSnapshot implements ToXContentFragment {
```

Comment:
`RemStoreBased` to `RemoteStoreBackedIndexShardMetadataSnapshot`. Also I think we can remove the prefix `BlobStore` as it is under that path. Seems redundant to me